### PR TITLE
feat: create a workflow for publishing new versions

### PR DESCRIPTION
**Pull request by omBratteng**
Not tested, will do that later

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          path: node_modules
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node_modules-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        if: steps.cache-node_modules.outputs.cache-hit != 'true'
+        run: |
+          yarn install --immutable
```
Comment (idoshamun) [inline .github/workflows/release.yml:47]: npm i guess?

Comment (omBratteng) [reply .github/workflows/release.yml:47]: 🙈

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+      - name: Setup git
+        run: |
+          git config --global tag.gpgsign true
+
+      - name: Bump version
+        run: |
+          npm version ${{ github.event.inputs.version }} --sign-git-tag
+
+      - name: Push changes
```
[the remark this answers is not in the document]
Comment (omBratteng) [reply .github/workflows/release.yml:57]: `npm version ${{ github.event.inputs.version }} --sign-git-tag` should create the tag, and then `git push --follow-tags` pushes commit _and_ tags

[the remark this answers is not in the document]
Comment (idoshamun) [reply .github/workflows/release.yml:57]: ah ok!

File: .github/workflows/release.yml (hunk 1 above)
Comment (omBratteng) [reply .github/workflows/release.yml]: copypasta fault, changed it to npm now

**Review by idoshamun — APPROVED** [review]

File: .github/workflows/release.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (omBratteng) [reply .github/workflows/release.yml:57]: FIY, this is how a commit with that command looks like
image

**Pull request merged**